pkg/core/blockchain.go

```diff
@@ -1326,7 +1356,7 @@ func (bc *Blockchain) isTxStillRelevant(t *transaction.Transaction, txpool *memp
 		return false
 	}
 	if txpool == nil {
-		if bc.dao.HasTransaction(t.Hash()) {
+		if bc.dao.HasTransaction(t.Hash()) != nil {
 			return false
 		}
 	} else if txpool.ContainsKey(t.Hash()) {
```

Comment:
I'd prefer to leave `ContainsKey` untouched and create a separate method for this, because there are other users of `ContainsKey` which are not always interested in transaction conflicts. And if I'm not mistaken, we already have suitable piece of code in mempool, now it's called `checkAttributesConflicts`. I added this check into `mp.RemoveStale` instead of adding it to `isTxStillRelevant` in order to avoid code duplication. Are we OK with that?

Reply:
Or are you talking about conflicts with dao here?

Reply:
This one is still relevant, conflicting transaction might get into the block, so you need to check mempool contents against it also (and this requires somehow using `txpool`).